### PR TITLE
Fix issue 657 - support for super hero protocol(estatsd) (WIP)

### PR DESCRIPTION
**Pull request by ljagiello**
Support for [super hero metrics](https://github.com/chef/stats_hero#metrics-reported-to-estatsd) ([estatsd](https://github.com/b/statsd_spec))

Extanding unit types based on https://github.com/chef/estatsd/blob/master/src/estatsd_shp.erl#L79-L86

WIP - metrics are collected correctly but there is "unknown" for me line eg.: `1|167`(https://github.com/chef/stats_hero#metrics-reported-to-estatsd)

### CODE REVIEW

**Comment by sparrc** [comment]
@ljagiello I don't think the metrics will actually get added on `Gather` with this, you need to add these two types to the `aggregate` function I believe?

File: plugins/inputs/statsd/statsd.go [hunk 1]
```diff
@@ -348,7 +348,7 @@ func (s *Statsd) parseStatsdLine(line string) error {
 				return errors.New("Error Parsing statsd line")
 			}
 			m.floatvalue = v
-		case "c", "s":
+		case "c", "s", "m", "mr":
```
Comment (sparrc) [inline plugins/inputs/statsd/statsd.go]: Are you sure these will never be floats? What are they anyways, do they act like counters or like gauges?

Comment (ljagiello) [inline plugins/inputs/statsd/statsd.go]: @sparrc waiting for more information about those - https://discourse.chef.io/t/super-hero-protocol-estatsd-format/7758

**Comment by ljagiello** [comment]
@sparrc you're absolutely right I'll fix `aggregate` part.

File: plugins/inputs/statsd/statsd.go (hunk 1 above)
Comment (ljagiello) [inline plugins/inputs/statsd/statsd.go]: `m` - a `meter` metric; increments by 1 (eg.: on each request)

```
1|478
bifrost.application.byStatusCode.200:1|m
```

https://github.com/chef/chef-server/blob/7c1df38aee7439674972a49bc157443b5c5c9cbf/src/oc_bifrost/doc/metrics.md#end-of-request

This one will be integer for sure.

**Comment by rijnhard** [comment]
It looks like this no longer works? Is this a regression or intentional?

**Comment by danielnelson** [comment]
@rijnhard Can you open a new issue with details of what you are experiencing?

**Pull request closed without merging**